### PR TITLE
Fix Bug - dropdown data.id mapping to element value and failing

### PR DESCRIPTION
**Pull request by yolochef**
**SUMMARY:**
Programmatically creating a drop-down list doesn't work. It creates a list, but no item reference can be obtained for any actions (such as onchange), as the current code tries to set the value attribute to the id of the data item. However, divs cannot have a value attribute, only select elements can have a value attribute.

**CODE CHANGE:**
data.value now maps to element value instead of to element id, and data.id maps to element id.

**BUG:**
When creating a dropdown on a div there are no ids assigned to the item elements. The value data parameter gets silently dropped, as divs don't have a value param.

This means that there is no way to know which element is selected by the user, other than using the data.name field, which is not a good design. 

**PROPOSAL:** 
I propose to add an id param to the item elements, which is a GLOBAL HTML attribute that every HTML element has. 

Good design patterns suggest that all select-able elements on the client must maintain ids that map to those in the database. Mapping can occur either directly or indirectly, on the client or on the server.

### CODE REVIEW

**Comment by fnchart** [comment]
This is a legacy breaking change, as it addresses a design-level issue. However, it would only break where the user used a element for their dropdown.

 I recommend we implement a temporary work-around the breaking change, and issue a warning in the JS console, and then remove the work-around in a later version. This will give users enough time to adapt. 

Let me know and I will revise the PR to accomodate this. We can even group all the various ones in a console group: 
‘’’
console.group("BREAKING CHANGES"); console.log("jSuites v2.3.2 -- DROPDOWN plugin has been deprecated and replaced. It will now use id in place of what value previously served. See PR #18 "); console.groupEnd(); 
‘’’

The only way to prevent this from breaking legacy code is if we split the logic between if it's related to a div or a select. However, that will cause all sorts of headaches down the line, and is VERY inadvisable. 

One thing to keep in mind is that most users may be happy with the change, and it probably won't be a big headache for them to fix.

**Comment by uwatr** [comment]
This is a legacy breaking change, as it addresses a design-level issue. However, it would only break where the user used a element for their dropdown.

 I recommend we implement a temporary work-around the breaking change, and issue a warning in the JS console, and then remove the work-around in a later version. This will give users enough time to adapt. 

Let me know and I will revise the PR to accomodate this. We can even group all the various ones in a console group: 
‘’’
`console.group("BREAKING CHANGES"); console.log("jSuites v2.3.2 -- DROPDOWN plugin has been deprecated and replaced. It will now use id in place of what value previously served. See PR #18 "); console.groupEnd(); `
‘’’

The only way to prevent this from breaking legacy code is if we split the logic between if it's related to a div or a select. However, that will cause all sorts of headaches down the line, and is VERY inadvisable. 

One thing to keep in mind is that most users may be happy with the change, and it probably won't be a big headache for them to fix.

**Comment by paulhodel** [comment]
Hi, sorry for the delay. I was working hard on new jexcel versions. Both CE & Pro. Now, I am dedicating time to bring jsuites to the next level. Firstly, thank you for your PR. But, on v3 the configuration of the item is different.

        // Create item
        var item = {}
        item.element = document.createElement('div');
        item.element.className = 'jdropdown-item';
        item.element.indexValue = obj.items.length;
        item.value = data.id;
        item.text = data.name;
        item.textLowerCase


So, basically the DOM is no longer the object itself, but now we keep the DOM in one of the properties. Following your suggestion partially, and for the sake of legacy, we can accept id or value to define the value of the item. Name or Label is accepted for the text shown to the user.

I am going to merge any code manually. ;)

The new version is much faster and have lazyloading, and would make a much faster rendering.

Regards,

**Comment by paulhodel** [comment]
jsuites v3 should be deployed next weekend.
